x-pack/plugins/enterprise_search/common/ml_inference_pipeline/index.ts

```diff
@@ -258,4 +260,7 @@ export const parseModelStateFromStats = (
 export const parseModelStateReasonFromStats = (trainedModelStats?: Partial<MlTrainedModelStats>) =>
   trainedModelStats?.deployment_stats?.reason;
 
-export const getMlInferencePrefixedFieldName = (fieldName: string) => `ml.inference.${fieldName}`;
+export const getMlInferencePrefixedFieldName = (fieldName: string) =>
+  `ml.inference.${stripMlInferencePrefix(fieldName)}`; // Strip first, then prepend, to prevent against double-prepending
+
+const stripMlInferencePrefix = (fieldName: string) => fieldName.replace('ml.inference.', '');
```

Comment:
nitpick: this will replace `ml.inference.` wherever it is in the string, not just at the start.